### PR TITLE
Use `datadog-ci` standalone binary instead of `npm` installed version

### PR DESCRIPTION
**Pull request by juan-fernandez**
### What does this PR do?
Use the standalone binary for `datadog-ci` for upload JUnit XML reports instead of using `npm`. In this case it isn't necessary because `npm` is available, but it is a way to test that the standalone binary works as it should. 

### Motivation
Dogfood the standalone binary for `datadog-ci`, which is useful for users that can't install node in their environment.

### CODE REVIEW

File: scripts/junit_report.js [hunk 1]
```diff
@@ -7,8 +7,6 @@ function uploadJUnitXMLReport () {
     console.log('Running in a fork. Skipping step.')
     return
   }
-  // we install @datadog/datadog-ci
-  execSync('yarn global add @datadog/datadog-ci@0.13.2', { stdio: 'inherit' })
```
Comment (rochdev) [inline scripts/junit_report.js:11]: I don't remember, why was this added in this way?

File: .circleci/config.yml [hunk 2]
```diff
@@ -35,7 +35,10 @@ commands:
           when: always
       - run:
           name: Upload JUnit XML Report
-          command: yarn junit:upload
+          command: |
+            curl -L "https://github.com/DataDog/datadog-ci/releases/latest/download/datadog-ci_linux-x64" --output "/usr/bin/datadog-ci"
```
Comment (rochdev) [inline .circleci/config.yml:39]: Wouldn't using this instead of the `npm` package make this more difficult to test locally? Or do we not support running this locally in the first place?

Comment (juan-fernandez) [reply .circleci/config.yml:39]: yeah it might not make much sense in this case.

File: scripts/junit_report.js (hunk 1 above)
Comment (juan-fernandez) [reply scripts/junit_report.js:11]: I think not to have it as a dependency

**Comment by juan-fernandez** [comment]
this change doesn't add anything to this repo so I'm dropping the PR
